build/charts/antrea/conf/antrea-controller.conf

```diff
@@ -71,3 +71,25 @@ nodeIPAM:
   # or when IPv6 Pod CIDR is not configured. Valid range is 64 to 126.
   nodeCIDRMaskSizeIPv6: {{ .nodeCIDRMaskSizeIPv6 }}
 {{- end }}
+
+ipsecCSRSigner:
+{{- with .Values.ipsec }}
+  enable: {{ eq .csrSigner.autoApprovePolicy "cert" }}
+  # Enable the CSR signer controller within the Antrea controller.
+  # Determines the auto approval policy of Antrea CSR signer for IPsec certificates management.
+  # It has the following options:
+  # never:                          Controller will never auto-approve CertificateSingingRequests and they need
+  #                                 to be approved manually by `kubectl certificate approve`
+  # validateAgentPodName (default): Controller will auto-approve the CertificateSingingRequest if the requestor
+  #                                 can be validated. This ensures that each Agent can only request certificates
+  #                                 for its own Node. This requires K8s `BoundServiceAccountTokenVolume` feature
+  #                                 to be enabled.
+  # always:                         Controller will auto-approve the CertificateSingingRequest without checking
+  #                                 the identity of the requestor.
+  autoApprovePolicy: {{ .csrSigner.autoApprovePolicy | quote }}
```

Comment:
I think "approvalPolicy" @antoninbas suggested sounds better, I actually considered the same name.

Reply:
I found the current approach is not safe and can be easily attacked as long as a attacker has CSR create permission. They can run a Pod on a Node to get a signed certificate of that Node since the approval only check whether the nodeName matches.

I think the approval should always check whether the CSR is created by antrea-agent by checking `spec.username` which is populated by the API server on creation and immutable.
Then for nodeName validating, it could just be nice to have as any attack means the service account is already compromised. On the other hand, there is no much value to let user configure "validateAgentPodName" or "always" because if the extra podName is not available, configuring "validateAgentPodName" will not work anyway (but it could log warnings about the risk and how to resolve it (upgrading to K8s >=1.22?), while if the extra podName is available, it's always correct to validate nodeName. Besides, "BoundServiceAccountTokenVolume" is a GA feature since 1.22 and cannot even be disabled, I think it's fine to ignore nodeName validation for clusters <=1.21.  Then there could be just one configuration: start auto approver or not.

@jianjuns @antoninbas @xliuxu thoughts?

Reply:
I thought we are checking agent name?

If we need not support a case agent is not running as a Pod, then sure we can just have enabled or not for >=1.21.

But for <=1.20, does controller have a way to know K8s version or BoundServiceAccountTokenVolume not enabled to ignore the Pod name validation? Do you mean if BoundServiceAccountTokenVolume is enabled, we can always know that from the token?

Reply:
Quan's proposal sounds good to me. I'm copying a CSR example shared by @xliuxu earlier at the end of this comment for easy reference.

We should always validate the authentication information under `spec.extra` when it is available and make sure it matches the nodeName. If the information is missing (because no `BoundServiceAccountTokenVolume`), we should log a warning as Quan suggested.

In our documentation, we should point out that for older K8s versions without `BoundServiceAccountTokenVolume`, we do not recommend auto-approval in production.

So +1 for having only 2 options for the approval policy.
We can use:
```
approvalPolicy: ["auto" | "manual"]
```
or
```
autoApproval: [true | false]
```

```yaml
apiVersion: certificates.k8s.io/v1
kind: CertificateSigningRequest
metadata:
  creationTimestamp: "2022-04-26T14:24:40Z"
  name: k8s-node-worker-1-ipsec
  resourceVersion: "112022"
  uid: fdab9d60-a7d0-4b72-9ac4-b3b991dfc7d3
spec:
  extra:
    authentication.kubernetes.io/pod-name:
    - antrea-agent-5vjjf
    authentication.kubernetes.io/pod-uid:
    - 3ede5c83-b42a-4f64-a325-a91d3db09d72
  groups:
  - system:serviceaccounts
  - system:serviceaccounts:kube-system
  - system:authenticated
  request: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURSBSRVFVRVNULS0tLS0KTUlJQ3BEQ0NBWXdDQVFBd01ERVNNQkFHQTFVRUNoTUpZVzUwY21WaExtbHZNUm93R0FZRFZRUURFeEZyT0hNdApibTlrWlMxM2IzSnJaWEl0TVRDQ0FTSXdEUVlKS29aSWh2Y05BUUVCQlFBRGdnRVBBRENDQVFvQ2dnRUJBTkt6CnlValVRZElRREFSd1NPSVhaUWFsSUtQRDZQd3YwMy9sNjd0amNtZXh6NmFSbzhGTjhnL09BSXZrbGRiTDQzMEgKaDFYZW5SMml1ekI3SFcwWjc0eUtsUHNldCtSc2YxZWMzbThQRk5vcTRoemd6ZEdBZDVqTnZyRnE4V0sxbVBmYQo3aG9mQ3FmSmlKT2dZcVNzZXdma3hVc1BEY0svVHRVay9nd29GOStLQXpyMXNXT01DeVpjTlQrUUxsdWFKZ1RLCkxnL0VuejFjTjJRWURyMDNWUEFhN0NWaU54LzJmY0xkdjRRTlJGcjAxSDAvNWJ5Z1l3Z01UbDNIZE1jQW53VzUKVjlBRk16NEdJMW1USldpR3k5NEFLVFNLTUxIVVpzcVEwVlFMTzFLSERicDRqaVVQRlFXZ3psYVk1bFpST0dQMgppUlBnd0FzUXFTRHg4TFhLWnhNQ0F3RUFBYUF2TUMwR0NTcUdTSWIzRFFFSkRqRWdNQjR3SEFZRFZSMFJCQlV3CkU0SVJhemh6TFc1dlpHVXRkMjl5YTJWeUxURXdEUVlKS29aSWh2Y05BUUVMQlFBRGdnRUJBTGQ2cUZpYWRNdEsKdFZwZ1dySTkxeWZyWjJSNmJLK3RxazhWZENxUm8rMmF0akt6VEVCRlVWendzaGtneVA4SUtVVUZrd2FCVmU0aApYcnhhTHU5a0NHVk5pYUxtVVRjbTl5QXN1L3FpaGc4dzNEYXZ4bzdGbENvVW9EMUViVG1FSjR5bVFCWlNlU1lCCkxFU29kU1R6MXovNHF1U0hEeEhhbWJKQzBsOWNqUUV1T3h4WEVWZ3Flb3h6cjlnQWpEUllrazBaSlR5cExINmsKdnhZVUlpbU42dzRDc0RGTnN4Wmx5TGt0ajdqR0tMd2RzRmtiTE1UbHJYQVRNaFRsRWNZaFh4ZWFvOWJ3NGs2bwppTGFnZk5nL1ZhdmMxUzNTVkd0VmpBVW1RdlM0YjRQY0xNaGFDemdSQ2wzQkRETldJcmpzYmh1b0szb3VsQ2RmCjNjMzI1czJoR09nPQotLS0tLUVORCBDRVJUSUZJQ0FURSBSRVFVRVNULS0tLS0K
  signerName: antrea.io/signer
  uid: aac69551-c616-4aaf-bea9-3f81aedaf7ba
  usages:
  - ipsec tunnel
  username: system:serviceaccount:kube-system:antrea-agent
status:
  certificate: LS0tLS1CRUdJTiBDRVJUSUZJQ0FURS0tLS0tCk1JSURPekNDQWlPZ0F3SUJBZ0lUQU4wOVJLY2VkMU5Tbm9kTnBFZkdyOGo1UkRBTkJna3Foa2lHOXcwQkFRMEYKQURBYU1SZ3dGZ1lEVlFRREV3OWhiblJ5WldFdGFYQnpaV010WTJFd0hoY05Nakl3TkRJMk1UUXlORE01V2hjTgpNekl3TkRJek1UUXlORFF3V2pBd01SSXdFQVlEVlFRS0V3bGhiblJ5WldFdWFXOHhHakFZQmdOVkJBTVRFV3M0CmN5MXViMlJsTFhkdmNtdGxjaTB4TUlJQklqQU5CZ2txaGtpRzl3MEJBUUVGQUFPQ0FROEFNSUlCQ2dLQ0FRRUEKMHJQSlNOUkIwaEFNQkhCSTRoZGxCcVVnbzhQby9DL1RmK1hydTJOeVo3SFBwcEdqd1UzeUQ4NEFpK1NWMXN2agpmUWVIVmQ2ZEhhSzdNSHNkYlJudmpJcVUreDYzNUd4L1Y1emVidzhVMmlyaUhPRE4wWUIzbU0yK3NXcnhZcldZCjk5cnVHaDhLcDhtSWs2QmlwS3g3QitURlN3OE53cjlPMVNUK0RDZ1gzNG9ET3ZXeFk0d0xKbHcxUDVBdVc1b20KQk1vdUQ4U2ZQVnczWkJnT3ZUZFU4QnJzSldJM0gvWjl3dDIvaEExRVd2VFVmVC9sdktCakNBeE9YY2QweHdDZgpCYmxYMEFVelBnWWpXWk1sYUliTDNnQXBOSW93c2RSbXlwRFJWQXM3VW9jTnVuaU9KUThWQmFET1Zwam1WbEU0ClkvYUpFK0RBQ3hDcElQSHd0Y3BuRXdJREFRQUJvMlF3WWpBVEJnTlZIU1VFRERBS0JnZ3JCZ0VGQlFjREJqQU0KQmdOVkhSTUJBZjhFQWpBQU1COEdBMVVkSXdRWU1CYUFGS0h3Z0V3YlB6UG1CMXB0VGdGZFRJR1V4TEgzTUJ3RwpBMVVkRVFRVk1CT0NFV3M0Y3kxdWIyUmxMWGR2Y210bGNpMHhNQTBHQ1NxR1NJYjNEUUVCRFFVQUE0SUJBUUEwCnArODYyRUZlY3lsOE9UWWlRdVpJRlJmM1JKYmF2VEE5VzZDK2ZFT2drT093ZFZGQjVuMXRZcGd1NHJGMnRmc3IKRmdTclVheG1yd1NHMmE1MXpWLzRiNjQ3cm8yVDF6aExXNWFvRW0zWW9qZlIxbTdROVgvYVd4Z0pyRm94di9SQgoremhWSFBKcnc1SWdwZ2lmTXI2NFhvMVhCUkVldDduMXpjalhKMXZDKzVOc2thODNBbGRjV0k5NjRXVnRRd25yClFaTFZ3amh3SVpDV2ZCaVlCd2llSEdKbWVaZmw0VU9jemdzL0p5QXJIR0JHa0txSmM5SW9EZUFQd3ZQRHRhbjQKVDdZWmlyWjRudU5SMHV2UFFhZXZ6MXFLWVQ0ZGVWOG1jZXBHMlZSN1hQY3d4dUEwcXNaWVZLWHdCU2FSN1R1bQpGTEhNbHgxc1hIemhIbzN2WDdzVgotLS0tLUVORCBDRVJUSUZJQ0FURS0tLS0tCg==
  conditions:
  - lastTransitionTime: "2022-04-26T14:24:40Z"
   lastUpdateTime: "2022-04-26T14:24:40Z"
    message: Automatically approved by antrea.io/signer
    reason: AutoApproved
    status: "True"
type: Approved
```

Reply:
Agree with @tnqn that we should also always check the `.sepc.username` to make sure the request is from antrea-agent ServiceAccount.
`approvalPolicy: ["auto" | "manual"]` is a good proposal. thanks!

Reply:
`autoApproval: [true | false]` may save some code `approvalPolicy` will have for declaring the constants and configuration parsing, if we don't foresee policies other than auto and manual.

Reply:
ok. will use `autoApproval: [true | false]` instead.